File: DevExpress-Demo/VB/PivotGridDemo.Wpf/ModuleResources/FieldViewModel.vb
Imports DevExpress.Mvvm.DataAnnotations
Imports DevExpress.Xpf.PivotGrid
Imports DevExpress.Mvvm.POCO

Namespace PivotGridDemo

    <POCOViewModel>
    Public Class FieldViewModel

        Public Shared Function Create(ByVal caption As String, ByVal name As String, ByVal area As FieldArea, ByVal areaIndex As Integer) As FieldViewModel
            Return ViewModelSource.Create(Function() New FieldViewModel(caption, name, area, areaIndex))
        End Function

        Protected Sub New(ByVal caption As String, ByVal name As String, ByVal area As FieldArea, ByVal areaIndex As Integer)
            Visible = True
            Me.Caption = caption
            Me.Name = name
            Me.Area = area
            Me.AreaIndex = areaIndex
        End Sub

        Public Overridable Property Area As FieldArea

        Public Overridable Property AreaIndex As Integer

        Public Overridable Property Caption As String

        Public Overridable Property GroupInterval As FieldGroupInterval

        Public Overridable Property Name As String

        Public Overridable Property SortByFieldName As String

        Public Overridable Property SortOrder As FieldSortOrder

        Public Overridable Property TopValueCount As Decimal

        Public Overridable Property TopValueShowOthers As Boolean

        Public Overridable Property Visible As Boolean

        Public Overridable Property Width As Double
    End Class
End Namespace

File: DevExpress-Demo/CS/PivotGridDemo.Wpf/ModuleResources/FieldViewModel.cs
using System;
using DevExpress.Mvvm.DataAnnotations;
using DevExpress.Xpf.PivotGrid;
using DevExpress.Mvvm.POCO;

namespace PivotGridDemo {
    [POCOViewModel]
    public class FieldViewModel {
        public static FieldViewModel Create(string caption, string name, FieldArea area, int areaIndex) {
            return ViewModelSource.Create(() => new FieldViewModel(caption, name, area, areaIndex));
        }
        protected FieldViewModel(string caption, string name, FieldArea area, int areaIndex) {
            Visible = true;
            Caption = caption;
            Name = name;
            Area = area;
            AreaIndex = areaIndex;
        }

        public virtual FieldArea Area { get; set; }
        public virtual int AreaIndex { get; set; }
        public virtual string Caption { get; set; }
        public virtual FieldGroupInterval GroupInterval { get; set; }
        public virtual string Name { get; set; }
        public virtual string SortByFieldName { get; set; }
        public virtual FieldSortOrder SortOrder { get; set; }
        public virtual Decimal TopValueCount { get; set; }
        public virtual bool TopValueShowOthers { get; set; }
        public virtual bool Visible { get; set; }
        public virtual double Width { get; set; }
    }
}
